lib/http-incoming.js

```diff
@@ -113,11 +113,25 @@ const PodiumHttpIncoming = class PodiumHttpIncoming {
         return this[_js];
     }
 
-    render(fragment) {
-        if (this.development) {
-            return this.view(fragment, this);
+    render(data) {
```

Comment:
I think I would prefer two arguments here:

```js
render(type, data) {
    if (type === 'podlet') {
        // do stuff for podlet
        return;
    }
    if (type === 'layout') {
        // do stuff for layout
        return;
    }
    throw new Error('"type" must be either "podlet" or "layout"');
}
```

This will in most cases be something which is dealt with by the plugin / wrapper for each http framework we are to support and by throwing if `type` is not defined is a way to yell at the developer of a plugin / wrapper that one have to distinguish on this when calling render.

Reply:
Something doesn't sit super well with me with the podlet and layout inside the http-incoming thing. I wonder if this should be handled in the layout and podlet classes somehow?

If we are always going to wrap the render method somewhere then perhaps we should handle the render or passthrough in the podlet which is the only place that will actually do that? That way the render method wouldn't need to worry about it at all.

Reply:
Something like this maybe?:

```js
// podlet
res.podiumSend = (data) => {
    if (this.development) {
        res.send(data.body);
        return;
    }
    res.send(res.locals.podium.render(data));
}

// layout
res.podiumSend = (data) => {
    res.send(res.locals.podium.render(data));
}
```

Reply:
or perhaps more like this:

```js
// utils
function render(data) {
  // ...
}

// podlet
const utils = require('@podium/utils');

class Podlet {
    ...

    render(data) {
        if (this.development) {
            return data;
        }
        return utils.render(data);
    }
}

// layout
const utils = require('@podium/utils');

class Layout {
    ...

    render(data) {
        return utils.render(data);
    }
}
```

Reply:
render should be on HttpIncoming in utils in my example above but you get the idea I'm going for

Reply:
I'm following your thinking. Though; the main reason for putting `.render()` on `HttpIncoming` was to cater for when one does not have http framework.

Examle:

```js
const HttpIncoming = require('@podium/util');
const Podlet = require('@podium/podlet');
const http = require('http');

const podlet = new Podlet({ [...snip...] })

const server = http.createServer(async (req, res) => {
    let incoming = new HttpIncoming(req, res);
    incoming = await podlet.process(incoming);
    res.end(incoming.render('<div>Jippi</div>'));
});
```

But I guess we can do this also:

```js
const HttpIncoming = require('@podium/util');
const Podlet = require('@podium/podlet');
const http = require('http');

const podlet = new Podlet({ [...snip...] })

const server = http.createServer(async (req, res) => {
    const incoming = new HttpIncoming(req, res);
    await podlet.process(incoming);
    res.end(podlet.render('<div>Jippi</div>'));
});
```

Reply:
The only difference the two is that in the last where the `.render()` method is on the podlet instance, we must pass the `HttpIncoming` object on to it if the template still should be able to read data from it.

In the first one do not need to pass the `HttpIncoming` on to the `.render()` method.

Reply:
ah right yea of course, I forgot render would still need access to HttpIncoming.